### PR TITLE
pennylane style is non-xkcd, pennylane_sketch is squiggly

### PR DESCRIPTION
**Pull request by timmysilv**
**Context:**
The PennyLane MPL-drawer style is great, but some users would prefer smoother plots for more professional contexts, eg. publications. We should continue to offer the xkcd-like style, but perhaps "pennylane" should be the non-xkcd style. I propose we make this change, and add "pennylane_sketch" to keep the squiggly, existing version.

**Description of the Change:**
- Rename the "pennylane" style to "pennylane_sketch"
- Introduce the "pennylane" style to be identical to the old style, but with non-sketch lines

**Benefits:**
A clean "pennylane" style for professional-looking data plotting, plus the sketch style is still offered.

**Possible Drawbacks:**
Would be breaking for the "pennylane" style. We can do a deprecation cycle if people prefer that - it's easy enough to add a warning, just lmk. I'd prefer not to add the warning, because then people will use the "pennylane" style that we recommend and get a warning in their face!

[sc-50495]

### CODE REVIEW

**Review by albi3ro — COMMENTED** [review]
Make sure to rerun https://github.com/PennyLaneAI/pennylane/blob/master/doc/_static/style/style_examples.py

File: doc/_static/style/solarized_light_style.png
Comment (trbromley) [inline doc/_static/style/solarized_light_style.png:1]: We don't have to fix it here, but isn't this colour scheme horrible? :laughing:

**Review by trbromley — COMMENTED** [review]
Thanks @timmysilv! I think it's ok to risk this breaking change, without a deprecation, given that `pennylane` is a plot style that's existed for just two releases and was not the default. The new `pennylane` style is also quite close to the old.

Please can we add this as a breaking change in the release notes?

CC @ikurecic.

File: doc/_static/style/solarized_light_style.png
Comment (timmysilv) [reply doc/_static/style/solarized_light_style.png:1]: tell that to [Ethan Schoonover](https://ethanschoonover.com/solarized/)! lol. afaik the solarized scheme is pretty popular in other parts of the computing world - I personally used to use solarized dark in vim

**Review by trbromley — APPROVED** [review]
Thanks @timmysilv!

File: doc/_static/style/solarized_light_style.png
Comment (albi3ro) [reply doc/_static/style/solarized_light_style.png:1]: Might look better if we bumped the contrast between lights and darks a bit more.

**Pull request merged**